// ngx-tinymce/esm2015/src/tinymce.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TinymceOptions } from './tinymce.options';
import { TinymceComponent } from './tinymce.component';
import * as i0 from "@angular/core";
export class NgxTinymceModule {
    static forRoot(options) {
        return {
            ngModule: NgxTinymceModule,
            providers: [{ provide: TinymceOptions, useValue: options }],
        };
    }
}
NgxTinymceModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxTinymceModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxTinymceModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxTinymceModule, declarations: [TinymceComponent], imports: [CommonModule], exports: [TinymceComponent] });
NgxTinymceModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxTinymceModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxTinymceModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [TinymceComponent],
                    exports: [TinymceComponent],
                }]
        }] });
